tests/test_preload_script.py

```diff
@@ -95,3 +95,332 @@ async def test_addPreloadScript_logging(websocket, context_id, html):
     # Assert navigation is finished.
     result = await read_JSON_message(websocket)
     assert result == {"id": command_id, "result": ANY_DICT}
+
+
+@pytest.mark.asyncio
+async def test_addPreloadScript_loadedInNewIframes_withCrossOrigin(
+        websocket, context_id, url_cross_origin, html):
+    await subscribe(websocket, "log.entryAdded")
+
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "expression": "() => console.log('my preload script')",
+                "context": context_id,
+            }
+        })
+
+    command_id = await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": html(),
+                "wait": "complete",
+                "context": context_id
+            }
+        })
+
+    # Log event should happen before navigation.
+    result = await read_JSON_message(websocket)
+    assert result == AnyExtending({
+        "method": "log.entryAdded",
+        "params": {
+            "args": [{
+                "type": "string",
+                "value": "my preload script"
+            }]
+        }
+    })
+
+    # Assert navigation is finished.
+    result = await read_JSON_message(websocket)
+    assert result == {"id": command_id, "result": ANY_DICT}
+
+    # Create a new iframe within the same context.
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""const iframe = document.createElement('iframe');
```

Comment:
I would add additional test when same-origin frame became OOPiF, to cover [this logic](https://github.com/GoogleChromeLabs/chromium-bidi/blob/618b4d0c09b2ecc47bf1278a91bc931781a8a44b/src/bidiMapper/domains/context/browsingContextProcessor.ts#L178).

Reply:
Resolving this comment and keeping track of it within the first comment of this PR as a task